src/ui/Controllers/CourseController.cs

```diff
@@ -14,25 +14,36 @@ namespace GovUk.Education.ManageCourses.Ui.Controllers
     [Route("organisation")]
     public class CourseController : CommonAttributesControllerBase
     {
-        private readonly ManageApi _manageApi;
+        private readonly IManageApi _manageApi;
 
-        public CourseController(ManageApi manageApi)
+        public CourseController(IManageApi manageApi)
         {
             _manageApi = manageApi;
         }
         [Route("{instCode}/course/{accreditingProviderId=self}/{ucasCode}")]
         public async Task<IActionResult> Variants(string instCode, string accreditingProviderId, string ucasCode)
         {
+            Validate(instCode, accreditingProviderId, ucasCode);
+
             var course = await _manageApi.GetCoursesByOrganisation(instCode);
 
             var providerCourse = "self".Equals(accreditingProviderId, StringComparison.InvariantCultureIgnoreCase)
                 ? course.ProviderCourses.First(c => String.IsNullOrEmpty(c.AccreditingProviderId))
                 : course.ProviderCourses
                     .First(c => c.AccreditingProviderId.Equals(accreditingProviderId, StringComparison.InvariantCultureIgnoreCase));
-            var courseDetail = providerCourse.CourseDetails.First(x => x.Variants.Select(v => v.UcasCode == ucasCode).Any());
+
+            var courseDetail = providerCourse.CourseDetails.FirstOrDefault(c => c.Variants.Any(v => v.UcasCode == ucasCode));
```

Comment:
Use `SingleOrDefault` if you only expect one.

https://enterprisecraftsmanship.com/2015/09/15/fail-fast-principle/

Reject.

Reply:
SingleOrDefault is now everywhere where only one is expected